### PR TITLE
Add swagger integration test

### PR DESCRIPTION
**Pull request by jimmidyson**
Fixes up static swagger generation and adds a basic integration test to ensure it doesn't revert again.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -311,7 +312,7 @@
           <version>${fabric8-maven-plugin.version}</version>
         </plugin>
         <plugin>
-          <groupId>com.github.kongchen</groupId>
+          <groupId>dk.nykredit.swagger</groupId>
```
Comment (rhuss) [inline pom.xml]: Just out of curiosity: Is this a different plugin or has the plugin been moved ? And it a change, what is the reason ?

File: runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    public void setRestTemplate(TestRestTemplate testRestTemplate) {
+        testRestTemplate.getRestTemplate().getMessageConverters().add(new YamlJackson2HttpMessageConverter());
+        this.restTemplate = testRestTemplate;
+    }
+
+    public TestRestTemplate restTemplate() {
```
Comment (rhuss) [inline runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java]: Still not keen on `get` methods ? ;-)

File: runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+    @Rule
+    public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    public void setRestTemplate(TestRestTemplate testRestTemplate) {
+        testRestTemplate.getRestTemplate().getMessageConverters().add(new YamlJackson2HttpMessageConverter());
```
Comment (rhuss) [inline runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java]: Why do we need a YAML converter ?

File: pom.xml (hunk 1 above)
Comment (jimmidyson) [inline pom.xml]: Previous plugin had problems with generating swagger document. New plugin actually works by simply delegating to Swagger's own libraries.

File: runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java (hunk 3 above)
Comment (jimmidyson) [inline runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java]: Oops thought I'd removed that - will do so now.

File: runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java (hunk 2 above)
Comment (jimmidyson) [inline runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java]: :) Fight!

File: runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java (hunk 3 above)
Comment (jimmidyson) [inline runtime/src/test/java/com/redhat/ipaas/runtime/BaseITCase.java]: Ah I remember, this is only used for swagger.yaml (YAML twin of swagger.json). Only needed for that test.

**Comment by jimmidyson** [comment]
[merge]

**Comment by jimmidyson** [comment]
Thanks @rhuss!

**Pull request merged**